File: heroes/src/main/java/com/caesar84mx/marvelheroes/screens/home/HeroesItemUIElement.kt
package com.caesar84mx.marvelheroes.screens.home

import com.caesar84mx.shared.common.data.model.Hero

data class HeroesItemUIElement(
    val name: String,
    val team: String,
    val publisher: String,
    val avatar: String
)

fun Hero.toUIElement(): HeroesItemUIElement = HeroesItemUIElement(
    name = name,
    team = team,
    publisher = publisher,
    avatar = avatar
)

File: ios/MarvelHeroes/Screens/Home/Data/HeroesItemUIElement.swift
//
//  HeroesItemUIElement.swift
//  MarvelHeroes
//
//  Created by Maxim Dymnov on 11/30/20.
//

import Foundation
import shared

struct HeroesItemUIElement: Identifiable {
    let id: UUID = UUID()
    let name: String
    let team: String
    let publisher: String
    let avatar: String
}

extension Hero {
    func toUiModel() -> HeroesItemUIElement {
        HeroesItemUIElement(
            name: name,
            team: team,
            publisher: publisher,
            avatar: avatar
        )
    }
}
